Update to blurse manager

Chr1Z93 opened the pull request:
- removed separate bless / curse tokens (bless / curse manager does all the handling - chaos token reserve still exists for "emergencies")
- removed the "take / remove" button since all cards have proper context menus ("Wendy Menu" exists for emeergencies)
- add token count labels
- added missing "Return" option to Nephthys
- renamed Wendy's menu
- added tracking for manually returned tokens
- added releasing when discarding cards
- some code cleanup

Entrox-Licher commented:
Excellent changes!  They are all working as intended.

This seems like a good opportunity to finish addressing the issues mentioned in #516 . Namely:

1. To add context menu support to the individual tokens for releasing seals.  This is essential as manual edits to the bag (be it manually removing or manually returning) full break this tool.  I still think tracking the tokens dynamically is best to avoid this edge case, but in absence of that (which would require a bigger rework), we should remove as many instances as possible that would prompt a user to manually interact with the bag, and this is the top offender

2. Make Wendy's Menu more obvious to users, for the same reason as (1).  Likely a good idea to rename it so it's function is more obvious (there are more corner cases than just Parallel Wendy), and definitely mention it explicitly in the manager's description.

3. If we are going to change the position of it, now would be the time to do that too (not necessary if we are opting to keep it where it is).  Maybe a copy of it below the chaos bag could make sense as best of both worlds talked about in the issue?

Chr1Z93 commented:
> Excellent changes! They are all working as intended.
> 
> This seems like a good opportunity to finish addressing the issues mentioned in #516 . Namely:
> 
> 1. To add context menu support to the individual tokens for releasing seals.  This is essential as manual edits to the bag (be it manually removing or manually returning) full break this tool.  I still think tracking the tokens dynamically is best to avoid this edge case, but in absence of that (which would require a bigger rework), we should remove as many instances as possible that would prompt a user to manually interact with the bag, and this is the top offender
> 2. Make Wendy's Menu more obvious to users, for the same reason as (1).  Likely a good idea to rename it so it's function is more obvious (there are more corner cases than just Parallel Wendy), and definitely mention it explicitly in the manager's description.
> 3. If we are going to change the position of it, now would be the time to do that too (not necessary if we are opting to keep it where it is).  Maybe a copy of it below the chaos bag could make sense as best of both worlds talked about in the issue?

1. I don't think this is as urgent as you make it out to be. There won't be any "full breaking", since you can use the existing context menu function to reset it's state (and dynamically get where tokens are at the moment).

2. For which cases would you need it outside of Parallel Wendy? Using custom card without scripting?

3. I'm still in the XML camp for this and might eventually have a small global window for it. Supporting multiple copies would require a bunch of code changes.

Entrox-Licher commented:
> > Excellent changes! They are all working as intended.
> > This seems like a good opportunity to finish addressing the issues mentioned in #516 . Namely:
> > 
> > 1. To add context menu support to the individual tokens for releasing seals.  This is essential as manual edits to the bag (be it manually removing or manually returning) full break this tool.  I still think tracking the tokens dynamically is best to avoid this edge case, but in absence of that (which would require a bigger rework), we should remove as many instances as possible that would prompt a user to manually interact with the bag, and this is the top offender
> > 2. Make Wendy's Menu more obvious to users, for the same reason as (1).  Likely a good idea to rename it so it's function is more obvious (there are more corner cases than just Parallel Wendy), and definitely mention it explicitly in the manager's description.
> > 3. If we are going to change the position of it, now would be the time to do that too (not necessary if we are opting to keep it where it is).  Maybe a copy of it below the chaos bag could make sense as best of both worlds talked about in the issue?
> 
> 1. I don't think this is as urgent as you make it out to be. There won't be any "full breaking", since you can use the existing context menu function to reset it's state (and dynamically get where tokens are at the moment).
> 2. For which cases would you need it outside of Parallel Wendy? Using custom card without scripting?
> 3. I'm still in the XML camp for this and might eventually have a small global window for it. Supporting multiple copies would require a bunch of code changes.

1. I'll admit the reset function is more robust than I originally thought, with the inclusion of the dynamic token detection.  There are still a lot of ways I can get it to break by just manually adding or removing tokens though (requiring multiple 'resets' in order to restore equilibrium, which is not very intuitive).  Removing all tokens does work in all cases I was able to test, but obviously this is somewhat of a nuclear option in actual play, where tokens will be sealed on all sorts of different cards.  I still think it's worth adding the functionality, since it is very likely to come up in play (Nephthys, Favor of the Sun/Moon, etc). 
2.  Custom cards are in fact the biggest scenario I was thinking of.  In general it is a good fallback option though, in order to prevent cards we may miss or support scenarios that could have similar effects.
3. I agree with this logic!  Let's keep it as is and we can revisit later if we do anything fancy XML-wise for the chaos bag.

4. Another bug I discovered: Wendy's Menu interacts very weirdly with cards that already have sealing options.  We should either make compatibility for that or (preferable I think) disable the interaction altogether (check for tag before adding Wendy's Menu)
